app/src/main/java/com/example/atheneum/viewmodels/FirebaseRefUtils/DatabaseWriteHelper.java

```diff
@@ -67,4 +68,58 @@ public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseR
     public static void deleteBook(User owner, Book book) {
         deleteBook(owner.getUserID(), book.getBookID());
     }
+
+    public static void makeRequest(Request request) {
+        HashMap<String, Object> updates = new HashMap<String, Object>();
+
+        final String requesterRef = String.format("requestCollection/%s/%s", request.getRequesterID(), request.getBookID());
+        final String bookRequestRef = String.format("bookRequests/%s/%s", request.getBookID(), request.getRequesterID());
+
+        updates.put(requesterRef, request);
+        updates.put(bookRequestRef, true);
+        RootRefUtils.ROOT_REF.updateChildren(updates, new DatabaseReference.CompletionListener() {
+            @Override
+            public void onComplete(@Nullable DatabaseError databaseError, @NonNull DatabaseReference databaseReference) {
+                if (databaseError != null) {
+                    Log.w(TAG, "Error updating data at " + databaseReference.toString());
+                    Log.i(TAG, "bookRequestRef: " + bookRequestRef.toString());
+                    Log.i(TAG, "requesterRef: " + requesterRef.toString());
+                } else {
+                    Log.i(TAG, "Successful update at " + databaseReference.toString());
+                }
+            }
+        });
+    }
+
+    public static void acceptRequest(Request request) {
+        HashMap<String, Object> updates = new HashMap<String, Object>();
+
+        final String requesterRef = String.format("requestCollection/%s/%s",
+                request.getRequesterID(), request.getBookID());
+        final String bookRequestRef = String.format("bookRequests/%s",
+                request.getBookID());
+        final String bookRef = String.format("books/%s/status", request.getBookID());
+

```

Comment:
Note that we have to decline all other requests when we accept one. See:

US 05.01.01
As an owner, I want to accept a request on one of my books. (Any other requests are declined.)

I wonder how that would work

Reply:
Maybe we don't really have to worry about that for now. Thoughts?

Reply:
According to what I was thinking, when the book is in ACCEPT state, all requests in the bookRequest field will just be replaced with requesterID.

Reply:
Oh wait. I think I know what you're getting at. We have to remove the bookID from the other requester's list. Maybe do DB check for every book in the requester's list for their status. If their status is ACCEPTED and their userID is not in the bookRequest book's field but we can remove it before rendering the data?

Reply:
But I don't think that change should be made for this particular PR since this PR is only concerned about the database changes.

Reply:
yes that's true and that may not even need to be done for this part as well but something to keep in mind for the future